### PR TITLE
NEW Hide iframe while loading in preview component

### PR DESCRIPTION
**Pull request by ScopeyNZ**
This is an arguable "fix". This allows components that render a preview to show an underlay or loading component while the iframe is loading.

### CODE REVIEW

File: client/src/components/Preview/Preview.js [hunk 1]
```diff
@@ -98,7 +114,12 @@ class Preview extends Component {
     }
 
     // Show iframe preview
-    return <iframe className="flexbox-area-grow preview__iframe" src={previewUrl} />;
+    return (<iframe
+      style={{ visibility: this.state.frameLoaded ? 'visible' : 'hidden' }}
+      className="flexbox-area-grow preview__iframe"
+      src={previewUrl}
+      onLoad={() => this.setState({ frameLoaded: true })}
```
Comment (robbieaverill) [inline client/src/components/Preview/Preview.js]: Do you want to make a method like `setFrameLoaded(loaded = true)` and use that here and line 29?

**Comment by robbieaverill** [comment]
I agree it's arguable, maybe retarget as `1` to be safe

File: client/src/components/Preview/Preview.js (hunk 1 above)
Comment (ScopeyNZ) [reply client/src/components/Preview/Preview.js]: Done

File: client/src/components/Preview/Preview.js [hunk 2]
```diff
@@ -12,9 +12,33 @@ class Preview extends Component {
...
+
+    if (prevProps.previewUrl !== this.props.previewUrl) {
+      this.setFrameLoaded(false);
+    }
+  }
+
+  /**
+   * Update state to indicate that the preview page has completed loading
+   *
+   * @param loaded
```
Comment (robbieaverill) [inline client/src/components/Preview/Preview.js]: Needs a type

File: client/src/components/Preview/Preview.js [hunk 3]
```diff
@@ -12,9 +12,33 @@ class Preview extends Component {
...
+   */
+  setFrameLoaded(loaded = true) {
```
Comment (robbieaverill) [inline client/src/components/Preview/Preview.js]: Do you need to bind this to the instance?

**Pull request merged**